scripts/sql/update.sql

```diff
@@ -0,0 +1,164 @@
+--
+-- event log to play ad-hoc sql
+-- relies on filename uniqueness to guarantee idempotency
+-- Usage: sudo -u postgres psql -d aatams -f update.sql -v path="'/home/meteo/aatams/scripts/sql'"
+--
+
+
+create or replace function exec_returning_string(s text) returns text
+language plpgsql volatile
+as $$
+  declare
+    ret text;
+  begin
+    execute s into ret;
+    return ret;
+  end;
+$$;
+
+
+create or replace function strip_comments(s text) 
+returns text
+language plpgsql
+as $$
+  declare
+    ret text;
+  begin
+    select regexp_replace( s, '^--[ \t-]+.*', '', 'gn') into ret;
+    return ret;
+  end;
+$$;
+
+
+create or replace function split_stmts(s text) 
+returns setof text
+language plpgsql
+as $$
+  declare
+    ret text;
+  begin
+    return query (select * from unnest(regexp_split_to_array(s, ';')));
```

Comment:
> Much better than json!

Not sure. I would have preferred json, but draw the line at having to encode my own line separator convention.

> Can you just add returning * to executed statements if they don't already have one rather than requiring them to be included?

I could - but feel it's probably clearer to code to the convention required for execute(), rather than have the code that executes the sql also manipulate that sql.